engine/flink/management/periodic/src/main/scala/pl/touk/nussknacker/engine/management/periodic/PeriodicProcessService.scala

```diff
@@ -58,6 +58,29 @@ class PeriodicProcessService(
 
   private implicit val localDateOrdering: Ordering[LocalDateTime] = Ordering.by(identity[ChronoLocalDateTime[_]])
 
+  def getScenarioActivitiesSpecificToPeriodicProcess(
+      processIdWithName: ProcessIdWithName
+  ): Future[List[ScenarioActivity]] =
+    scheduledProcessesRepository
+      .getSchedulesState(processIdWithName.name)
+      .run
+      .map(_.groupedByPeriodicProcess)
+      .map(_.flatMap(_.deployments))
+      .map(_.map { deployment =>
+        ScenarioActivity.PerformedScheduledExecution(
+          scenarioId = ScenarioId(processIdWithName.id.value),
+          scenarioActivityId = ScenarioActivityId.random,
+          user = ScenarioUser(None, UserName("Nussknacker"), None, None),
+          date = deployment.createdAt.toInstant(ZoneOffset.UTC),
+          scenarioVersionId = Some(ScenarioVersionId(deployment.periodicProcess.processVersion.versionId.value)),
+          dateFinished = deployment.state.completedAt.map(_.toInstant(ZoneOffset.UTC)),
+          scheduleName = deployment.scheduleName.display,
+          status = deployment.state.status.toString,
```

Comment:
why do we have to use "toString"?

Reply:
Refactored, we now have sealed trait in the ScenarioActivity domain, which represents statuses. (It was toString before, because this value is Scala enumeration)

Other change:
Activity's "date" changed from periodic "createdAt" to "runAt"; I noticed it in tests, the correct date of the activity should be the date of execution, not o scheduling, IMO it was my mistake; added field "createdAt"